bridges/RedditBridge.php

```diff
@@ -27,14 +27,32 @@ class RedditBridge extends FeedExpander {
 
 	public function collectData(){
 
-		switch($this->queriedcontext) {
+		switch($this->queriedContext) {
 			case 'single': $subreddits[] = $this->getInput('r'); break;
 			case 'multi': $subreddits = explode(',', $this->getInput('rs')); break;
 	}
 
 		foreach ($subreddits as $subreddit) {
 			$name = trim($subreddit);
-			$this->collectExpandableDatas("https://www.reddit.com/r/$name/.rss");
+			$url = "https://www.reddit.com/r/$name/.rss";
+
+			// If in CLI mode with no root certificates defined, skip the url verification
+			if(php_sapi_name() !== 'cli' || !empty(ini_get('curl.cainfo'))) {
+				// We must test if the subreddit exists before gathering the content
+				$ch = curl_init();
+				curl_setopt($ch, CURLOPT_URL, $url);
+				curl_setopt($ch, CURLOPT_USERAGENT, ini_get('user_agent'));
+				curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+				curl_exec($ch);
+				$httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+				curl_close($ch);
+
+				if($httpcode !== 403 && $httpcode !== 302) {
+					$this->collectExpandableDatas($url);
+				}
+			} else {
+				$this->collectExpandableDatas($url);
+			}
```

Comment:
Referring to your last comment: https://github.com/RSS-Bridge/rss-bridge/pull/1213#issuecomment-548778066

RSS-Bridge is supposed to return an error if the resource doesn't exist. This is the intended way to let users know that they provided incorrect input data. In either case, a bridge should never circumvent the API. If you think it doesn't work as intended, don't hesitate to open an issue for further discussion.

With the `queriedContext` fix from above, my machine correctly responds with

```text
Reddit Bridge was unable to receive or process the
remote website's content!

Error message: `The requested resource cannot be found!

Please make sure your input parameters are correct!

cUrl error:  (0)

PHP error: `

Query string: `action=display&bridge=Reddit&context=single&r=react&format=Atom`

Version: `git.master.3bc8c94`
```

Am I missing something?